packages/patterns/integration/shadow-dom-utils.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * Utility functions for navigating through shadow DOM boundaries in tests
+ */
+
+type Page = any; // The Page type from integration utils
```

Comment:
Can this be `@astral/astral#Page`? would apply to both Page and our `@commontools/integration` wrapper